syft/pkg/cataloger/java/parse_pom_xml.go

```diff
@@ -109,3 +113,11 @@ func cleanDescription(original string) (cleaned string) {
 	}
 	return strings.TrimSpace(cleaned)
 }
+
+func propertyReplacer(pomProperties gopom.Properties, stringWithVariablesToReplace string) string {
+	var propertyMatcher = regexp.MustCompile("[$][{][^}]+[}]")
```

Comment:
Quick note: this would be maybe more idiomatic go to initialize a variable in a function like `propertyMatcher := regexp.MustCompile("[$][{][^}]+[}]")` (instead of the `var` definition). However, this can be moved out of the function so the regex is only compiled once, e.g.:
```go
var propertyMatcher = regexp.MustCompile("[$][{][^}]+[}]")

func propertyReplacer(pomProperties gopom.Properties, stringWithVariablesToReplace string) string {
```